internal/services/securitycenter/security_center_subscription_pricing_resource.go

```diff
@@ -134,23 +134,53 @@ func resourceSecurityCenterSubscriptionPricingUpdate(d *pluginsdk.ResourceData,
 	}
 
 	extensionsStatusFromBackend := make([]pricings_v2023_01_01.Extension, 0)
-	if err == nil && apiResponse.Model != nil && apiResponse.Model.Properties != nil && apiResponse.Model.Properties.Extensions != nil {
-		extensionsStatusFromBackend = *apiResponse.Model.Properties.Extensions
+	isCurrentlyInFree := false
+	if err == nil && apiResponse.Model != nil && apiResponse.Model.Properties != nil {
+		if apiResponse.Model.Properties.Extensions != nil {
+			extensionsStatusFromBackend = *apiResponse.Model.Properties.Extensions
+		}
+
+		if apiResponse.Model.Properties.PricingTier == "Free" {
+			isCurrentlyInFree = true
+		}
 	}
 
 	if vSub, okSub := d.GetOk("subplan"); okSub {
 		pricing.Properties.SubPlan = utils.String(vSub.(string))
 	}
-	if d.HasChange("extension") || d.IsNewResource() {
+	if d.HasChange("extension") {
 		// can not set extensions for free tier
 		if pricing.Properties.PricingTier == pricings_v2023_01_01.PricingTierStandard {
 			var extensions = expandSecurityCenterSubscriptionPricingExtensions(d.Get("extension").(*pluginsdk.Set).List(), &extensionsStatusFromBackend)
 			pricing.Properties.Extensions = extensions
 		}
 	}
 
-	if _, err := client.Update(ctx, id, pricing); err != nil {
-		return fmt.Errorf("setting %s: %+v", id, err)
+	if pricing.Properties.PricingTier == "Free" && pricing.Properties.Extensions != nil && len(*pricing.Properties.Extensions) > 0 {
+		return fmt.Errorf("Can not have enabled extensions when switching to free tier, please remove the extensions")
+	}
```

Comment:
```suggestion
	if len(d.Get("extension").(*pluginsdk.Set).List()) > 0 && pricing.Properties.PricingTier == pricings_v2023_01_01.PricingTierFree {
		return fmt.Errorf("extensions cannot be enabled when using free tier")
	}
```

We should change this check to specifically what has been input in the config and move this line above the `if d.HasChange("extension") {` check on line 151, as a user could change from Standard -> Free without making any changes to extensions, meaning this would get missed. It also means this will check will work when a user attempts to create the resource with `Free` tier and extensions enabled.

Reply:
thanks for the suggestion